projects/plugins/jetpack/extensions/blocks/subscriber-login/subscriber-login.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Subscriber Login Block.
+ *
+ * @since 0.0.1
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Extensions\Subscriber_Login;
+
+use Automattic\Jetpack\Blocks;
+use Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Abstract_Token_Subscription_Service;
+use Automattic\Jetpack\Status\Host;
+use Jetpack_Gutenberg;
+use Jetpack_Memberships;
+use Jetpack_Options;
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	if ( ! \Jetpack::is_module_active( 'subscriptions' ) || ! class_exists( '\Jetpack_Memberships' ) ) {
+		return;
+	}
+
+	Blocks::jetpack_register_block(
+		__DIR__,
+		array( 'render_callback' => __NAMESPACE__ . '\render_block' )
+	);
+
+	add_action( 'wp_logout', __NAMESPACE__ . '\subscriber_logout' );
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Logout subscriber.
+ *
+ * @return void
+ */
+function subscriber_logout() {
+	Abstract_Token_Subscription_Service::clear_token_cookie();
+}
+
+/**
+ * Returns current URL.
+ *
+ * @return string
+ */
+function get_current_url() {
```

Comment:
There might be existing Jetpack helper for this already.

Reply:
You're right. Will use it in a follow-up PR.
https://github.com/Automattic/jetpack/blob/20e0d8fe09b53c71910a9e31e68b28d7618ac6b0/projects/packages/boost-core/src/lib/class-url.php#L39

Reply:
Another https://github.com/Automattic/jetpack/blob/20e0d8fe09b53c71910a9e31e68b28d7618ac6b0/projects/packages/publicize/src/class-share-limits.php#L82

Not sure how to make those available for jetpack plugin, since they're from different project/package in the monorepo.

Reply:
@pkuliga Here's a good one to use since it's already in subscribe block:

https://github.com/Automattic/jetpack/blob/bd4b3ed9532d80ab495fbe0f59e8c0c2ac141d84/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L909-L939